lib/buildkite/pipelines/helpers/retry.rb

```diff
@@ -4,15 +4,36 @@ module Buildkite
   module Pipelines
     module Helpers
       module Retry
-        def automatic_retry_on(exit_status:, limit:)
+        def automatic_retry_on(options = {})
```

Comment:
Is there a reason to move away from keyword args? I think you might end up with deprecation warnings when trying to use it as keywords in other places (ie like when upgrading to ruby 3).

If you want to make them optional, add a default value of nil:

```suggestion
        def automatic_retry_on(exit_status: nil, limit: nil, signal_reason: nil)
```

I think this approach has a bit better developer experience, because you get errors when trying to pass keywords arguments that aren't supported.